src/main/java/org/reaktivity/nukleus/oauth/internal/OAuthRealms.java

```diff
@@ -66,40 +71,58 @@ public OAuthRealms(
     private OAuthRealms(
         Map<String, JsonWebKey> keysByKid)
     {
-        keysByKid.forEach((k, v) -> add(v.getKeyId()));
         this.keysByKid = keysByKid;
     }
 
-    public void add(
-        String realm)
+    public long resolve(
+        String realmName,
+        String[] scopeNames)
     {
-        if (realmsIdsByName.size() == MAX_REALMS)
+        try
         {
-            throw new IllegalStateException("Too many realms");
+            final OAuthRealm realm = realmsIdsByName.computeIfAbsent(realmName, this::newOAuthRealm);
+            return realm.resolve(scopeNames);
+        }
+        catch(IllegalStateException ise)
+        {
+            return NO_AUTHORIZATION;
         }
```

Comment:
How about we eliminate the exception flow by changing the code as follows:
```
    long authorization = NO_AUTHORIZATION;
    if (nextRealmBit < MAX_REALMS)
    {
        final OAuthRealm realm = realmsIdsByName.computeIfAbsent(realmName, this::newOAuthRealm);
        authorization = realm.resolve(scopeNames);
    }
    return authorization;
```
... and change the check in `newOAuthRealm` to an `assert`.

Note: we'll also need a companion change in `OAuthRealm.resolve` method.